### DISCUSSION TITLE
Adding python/django path

### DISCUSSION THREAD

**Discussion by shivakharbanda** [TheOdinProject/curriculum#28724] [Ideas]
### Checks

- [x] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

Introducing a Python/Django path alongside Node.js could greatly benefit and enhance the breadth of The Odin Project's curriculum.

### Path

Other / NA

### Lesson Url

https://www.theodinproject.com/lessons/

**Comment by shivakharbanda** [comment]
If it's permitted, I would like to participate and forge a path.

**Comment by bycdiaz** [comment]
@shivakharbanda Can you share more about how you feel this would benefit and enhance the curriculum?

**Comment by shivakharbanda** [comment]
Incorporating a Python/Django stack into the curriculum could provide significant benefits to students. Python, particularly when paired with Django, is exceptionally well-suited for AI/ML applications, which are rapidly becoming essential skills in the tech industry. By offering students the opportunity to learn Python/Django, The Odin Project can empower them to build a more diverse range of applications, enhancing their project portfolios. This, in turn, could increase their job prospects by making them more versatile developers who are proficient in both front-end and back-end development across multiple languages and frameworks.

Adding this option would not only enrich the learning experience but also better prepare students for the evolving demands of the tech job market.

**Comment by JoshDevHub** [comment]
This won't be a priority for our team for a few different reasons.
1. It'd be very difficult to build out a full path for Python/Django. I'd estimate the project for getting all the required content written would take around a year, and we'd need multiple contributors working on it. You alone will not be enough, and I'm not quite sure you appreciate how much an investment this will be.
2. Even after the content exists, we'd need people to help maintain the path as Python, Django, and the surrounding ecosystem for those technologies will quickly evolve over time. Once again, you can't do this alone, and there are (to my knowledge) no members of our current maintainer team who regularly work with Python or Django.
3. Python, being a dynamically typed interpreted language, is very similar to the languages we already teach in the curriculum (Ruby and JavaScript). Django is a fullstack MVC framework that is very similar to Ruby on Rails, so much so that you will often find Django job postings that say they want candidates with "experience in Django, Laravel, Rails or some other MVC framework." I'm struggling to see the value of introducing something that'd already be fairly comfortable for TOP learners to transition into if they need it. You mention ML and AI, and those are potentially compelling reasons to get involved with Python. But we're currently not interested in growing the curriculum beyond web app development for similar reasons as points 1 and 2 in this list. Same reason we don't have courses on game dev, kernel dev, or mobile dev. We just don't have the required group of maintainers to support such things.
